state/cleanup.go

```diff
@@ -865,7 +866,7 @@ func (st *State) cleanupUnitsForDyingApplication(
 		}
 	}
 	for _, u := range unitsToDestroy.Values() {
-		err = applicationService.DestroyUnit(ctx, u)
+		err = applicationService.DestroyUnit(ctx, coreunit.Name(u))
```

Comment:
**todo:** Unvalidated name: at least add a comment to explain why it is safe to not use NewName here. If it is not safe, validate the name.

Reply:
fair enough (general comment)